src/ra_log.erl

```diff
@@ -139,7 +141,7 @@ init(#{uid := UId} = Conf) ->
             [File | _] ->
                 %% TODO provide function that only reads the index and term
                 %% of the snapshot file.
-                {ok, {SI, ST}} = ra_log_snapshot:read_indexterm(File),
+                {ok, {SI, ST}} = SnapModule:read_indexterm(File),
```

Comment:
module calls like this should go through the behaviour they are implemented: `ra_snapshot(SnapModule, File)` - This helps validating these calls can keep the dynamic dispatch separated from "normal" code